lib/utils.js

```diff
@@ -4,60 +4,62 @@ import { exec } from 'teen_process';
 import { waitForCondition } from 'asyncbox';
 import { getVersion } from 'appium-xcode';
 import { getDevices } from 'node-simctl';
-import { fs, system } from 'appium-support';
+import { fs } from 'appium-support';
 import { Certificate } from './certificate';
 import path from 'path';
 import Simulator from './simulator-xcode-6';
 
 
-const OSASCRIPT_TIMEOUT = 10000;
+const DEFAULT_SIM_SHUTDOWN_TIMEOUT = 10000;
 
-async function sendKillAll (appName) {
+// pgrep/pkill exit codes:
+// 0       One or more processes were matched.
+// 1       No processes were matched.
+// 2       Invalid options were specified on the command line.
+// 3       An internal error occurred.
+
+async function forceKillAll (appName) {
   try {
-    await exec('killall', [appName]);
-  } catch (err) {
-    // if no process exists we'll get an error that we expect.
-    // otherwise throw it again
-    if (err.stderr.indexOf(`No matching processes belonging to you were found`) === -1) {
-      throw err;
+    await exec('pkill', ['-9', '-x', appName]);
+  } catch (e) {
```

Comment:
@imurchie I'm not 100% sure here about teen_process lib functionality. Does it throw an exception if the exit code differs from zero?